src/yacs-web/src/components/SelectedCourses.vue

```diff
@@ -0,0 +1,153 @@
+                    class="selected-course-card"
+                >
+                    <button type="button" class="close text-muted" @click="removeCourse(course)">
+                        <span>&times;</span>
+                    </button>
+                    {{readableDate(course.date_start)}}-{{readableDate(course.date_end)}}
+
+                    <b-list-group flush>
+                        <b-list-group-item
+                            button
+                            v-for="section in course.sections"
+                            :key="section.crn"
+                            @click.stop="toggleCourseSection(course, section)"
+                            :style="{
+                                'border-left': section.selected ? `4px solid ${ColorService.getBorderColor(section)}` : 'none',
+                                'background-color': section.selected ? `${ColorService.getBackgroundColor(section)}` : 'white'
+                            }"
+                        >
+                            {{section.crn}} - {{section.sessions[0].section}}<br>
+
+                            <span
+                                v-for="courseSession in section.sessions"
+                                :key="courseSession.crn + courseSession.day_of_week + courseSession.time_start"
+                            >
+                                {{DAY_SHORTNAMES[courseSession.day_of_week + 1]}}: {{readableTime(courseSession.time_start)}} - {{readableTime(courseSession.time_end)}}
+                                <br>
+                            </span>
+
+                        </b-list-group-item>
+                    </b-list-group>
+                </b-card>
+            </b-card-group>
+        </b-col>
+    </b-row>
+</template>
+
+<script>
+import { 
+    TimeDateMixin 
+} from '@/mixins';
+
+import {
+    ColorService
+} from '@/services';
+
+import { faPaperPlane } from '@fortawesome/free-solid-svg-icons';
+
+export default {
+    name: 'SelectedCourses',
+    mixins: [
+        TimeDateMixin
+    ],
+    props: {
+        courses: Object,
+    },
+    data () {
+        return {
+            ColorService,
+            exportIcon: faPaperPlane,
+            ICS_DAY_SHORTNAMES: ["MO", "TU", "WE", "TH", "FR", "SA", "SU"],
+        }
+    },
+    methods: {
+        removeCourse (course) {
+            this.$emit('removeCourse', course);
+        },
+        toggleCourseSection (course, section) {
+            if (section.selected) {
+                this.$emit('removeCourseSection', section);
+            } else {
+                // Only allow selection of one section per course
+                this.$emit('removeAllCourseSections', course);
+                this.$emit('addCourseSection', course, section);
+            }
+        },
+        exportScheduleToIcs () {
+            let calendarBuilder = window.ics()
+            let semester;
+
+            for (const course of Object.values(this.courses)) {
```

Comment:
The reason I turned `this.courses` into a dictionary was so that it was easy and fast to get the course information of a given session.

However, when looping over all courses and figuring out the matching section, it kind of eliminates the need for `courses` to be a dictionary. `_getCourseIdentifier` doesn't really serve a purpose anymore. I struggle to think if there would be an eventual need.

Reply:
I like the concept of generating ids and the dictionary and proposed having this done on the backend. At the moment, since the data is nested, access to the children elements are generally within the scope of the parent, so the backpointer's are usually not necessary. However, the id is definitely useful for normalized data (no nesting) which is still an option we may explore in the future. 
Additionally, object properties now are ordered in modern browsers + ES6 so using them as arrays should be fine. So we can probably leave this as it is and in the case we need to use them or decide to normalize the data they will come in handy.